src/frontend/src/lib/utils/transactions.utils.ts

```diff
@@ -95,8 +106,24 @@ export const mapAllTransactionsUi = ({
 		}
 
 		if (isNetworkIdICP(networkId)) {
-			// TODO: Implement ICP transactions
-			return acc;
+			// TODO: Implement ckBTC and ckETH pending transactions
```

Comment:
Unsolicited feedback: the more you provide PR for this feature, the more I think it's unrealisitc in terms of performance but, let's see.

Reply:
yes... the final version is not exactly good; we should really start thinking about caching